### PR TITLE
added DEVELOPMENT.md for knative-gcp

### LINKED ISSUE #1463: Create DEVELOPMENT.md for knative-gcp
**Problem**
Create DEVELOPMENT.md for knative-gcp

**Exit Criteria**
A measurable (binary) test that would indicate that the problem has been resolved.
There is a DEVELOPMENT.md for knative-gcp

### PR DESCRIPTION
**Pull request by capri-xiyue**
Fixes #1463 

## Proposed Changes

- 🎁 Added DEVELOPMENT.md for knative-gcp
- 🎁 Updated installation of installing-knative-GCP.md

### CODE REVIEW

File: DEVELOPMENT.md [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+  must run [`./hack/update-deps.sh`](./hack/update-deps.sh).
+
+These are both idempotent, and we expect that running these at `HEAD` to have no
+diffs.
+
+Once the codegen and dependency information is correct, redeploying the
+controller is simply:
+
+```shell
+ko apply -f config/controller.yaml
```
Comment (nachocano) [inline DEVELOPMENT.md:62]: I think this should be 500-controller.yaml

File: DEVELOPMENT.md [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+This doc explains how to setup a development environment so you can get started
+contributing to `Knative-GCP`.
+Also take a look at:
+
+- [The pull request workflow](https://www.knative.dev/contributing/contributing/#pull-requests)
+- [How to add and run e2e tests](./test/e2e/README.md)
+- [Iterating](#iterating)
+- [Format check](#format-check)
+
+## Getting started
```
Comment (nachocano) [inline DEVELOPMENT.md:12]: I think we might need: https://github.com/knative/eventing/blob/master/DEVELOPMENT.md#requirements
and also https://github.com/knative/eventing/blob/master/DEVELOPMENT.md#setup-your-environment here

File: DEVELOPMENT.md [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+
+## Iterating
+
+As you make changes to the code-base, there are two special cases to be aware
+of:
+
+- **If you change a type definition ([pkg/apis/](./pkg/apis/.)),** then you must
+  run [`./hack/update-codegen.sh`](./hack/update-codegen.sh).
+- **If you change a package's deps** (including adding external dep), then you
+  must run [`./hack/update-deps.sh`](./hack/update-deps.sh).
```
Comment (grac3gao) [inline DEVELOPMENT.md:53]: We used to have ` ./hack/../vendor/knative.dev/test-infra/scripts/library.sh: line 29: conditional binary operator expected` error when running `./hack/update-codegen.sh` in Mac, the root cause is bash version in Mac needs to upgrade (the default version for bash is dated from 2007...). Might worth to indicate this here.

File: DEVELOPMENT.md (hunk 2 above)
Comment (capri-xiyue) [reply DEVELOPMENT.md:12]: This part is included in https://github.com/google/knative-gcp#installing-knative-gcp which is linked below in this doc

File: DEVELOPMENT.md (hunk 3 above)
Comment (capri-xiyue) [reply DEVELOPMENT.md:53]: done

**Pull request merged**